From: Wengling Chen <feiyu2817@gmail.com>
Date: Sat, 2 Nov 2019 07:56:44 +0100
Subject: Remove dependency on com.google.android.gms.gcm

It also disalbes Snippets, auto-update of offline pages and launching browser from background service.
The snippets are already disabled in native codes, this will disable it on Java level.
---
  chrome/android/BUILD.gn                      |  2 --
  chrome/android/javatests/BUILD.gn            |  1 -
  chrome/android/junit/BUILD.gn                |  1 -
  third_party/android_deps/autorolled/BUILD.gn | 19 -------------------
  4 files changed, 23 deletions(-)

--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -315,7 +315,6 @@ if (_is_default_toolchain) {
       "$google_play_services_package:google_play_services_basement_java",
       "$google_play_services_package:google_play_services_cast_framework_java",
       "$google_play_services_package:google_play_services_cast_java",
-      "$google_play_services_package:google_play_services_gcm_java",
       "$google_play_services_package:google_play_services_iid_java",
       "$google_play_services_package:google_play_services_tasks_java",
       "$google_play_services_package:google_play_services_vision_common_java",
@@ -2009,7 +2008,6 @@ if (_is_default_toolchain) {
     ]
     deps = [
       ":chrome_java",
-      "$google_play_services_package:google_play_services_gcm_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//build/android:build_java",
--- a/chrome/android/javatests/BUILD.gn
+++ b/chrome/android/javatests/BUILD.gn
@@ -55,7 +55,6 @@ android_library("chrome_test_java_helper
     "$google_play_services_package:google_play_services_basement_java",
     "$google_play_services_package:google_play_services_cast_framework_java",
     "$google_play_services_package:google_play_services_cast_java",
-    "$google_play_services_package:google_play_services_gcm_java",
     "$google_play_services_package:google_play_services_iid_java",
     "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
--- a/chrome/android/junit/BUILD.gn
+++ b/chrome/android/junit/BUILD.gn
@@ -51,7 +51,6 @@ if (is_android) {
       "$google_play_services_package:google_play_services_basement_java",
       "$google_play_services_package:google_play_services_cast_framework_java",
       "$google_play_services_package:google_play_services_cast_java",
-      "$google_play_services_package:google_play_services_gcm_java",
       "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:base_java_test_support",
--- a/third_party/android_deps/autorolled/BUILD.gn
+++ b/third_party/android_deps/autorolled/BUILD.gn
@@ -253,25 +253,6 @@ template("define_autorolled_targets") {
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
     if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_gcm_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_gcm/play-services-gcm.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_gcm/com_google_android_gms_play_services_gcm.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_iid_java",
-          "$google_play_services_package:google_play_services_stats_java",
-          "//third_party/androidx:androidx_collection_collection_java",
-          "//third_party/androidx:androidx_core_core_java",
-          "//third_party/androidx:androidx_legacy_legacy_support_core_utils_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
       android_aar_prebuilt("google_play_services_identity_credentials_java") {
         aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_identity_credentials/play-services-identity-credentials.aar"
         info_path = "autorolled/committed/libs/com_google_android_gms_play_services_identity_credentials/com_google_android_gms_play_services_identity_credentials.info"
